transport/http/transport.go

```diff
@@ -50,6 +54,16 @@ type TransportOption func(*transportConfig)
 
 func (TransportOption) httpOption() {}
 
+// NilOption is empty option - example use case:
```

Comment:
Here's our use case:

```
    topt := http.NilOption()
    if proxy != nil && proxy.Socks5 != nil {
        topt = http.Socks5Proxy(*proxy.Socks5)
    }
    dispatcher := yarpc.NewDispatcher(
        yarpc.Config{
            Name: "peloton-outbound",
            Outbounds: yarpc.Outbounds{
                pelotonAPIService: transport.Outbounds{
                    Unary: http.NewTransport(topt).NewSingleOutbound(murl),
                },
                pelotonResMgr: transport.Outbounds{
                    Unary: http.NewTransport(topt).NewSingleOutbound(rurl),
                },
            },
        })
```
If I didn't have Nil option I'd have to have two separate sections for creation - one if no options and one with the Socks5Proxy.